polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -776,8 +785,15 @@ trait ValidationBackend {
 		// long.
 		let total_time_start = Instant::now();
 
-		let mut validation_result =
-			self.validate_candidate(pvf.clone(), exec_timeout, params.encode()).await;
+		// Use `Priority::Critical` as finality trumps parachain liveliness.
+		let mut validation_result = self
+			.validate_candidate(
+				pvf.clone(),
+				exec_timeout,
+				params.encode(),
+				polkadot_node_core_pvf::Priority::Critical,
```

Comment:
Nah. This is way to implicit. The docs of `validate_candidate_with_retry` say that is is used in approvals, but this is just an assumption about its usage, one the might even make sense to remove. The docs should focus on what the function does and the caller then decides whether it is suitable for the job at hand.

TL;DR: This function should take a priority argument, instead of assuming calling context.

Reply:
yeah it makes sense